followthemoney/types/date.py

```diff
@@ -95,3 +95,23 @@ def compare(self, left, right):
 
     def rdf(self, value):
         return Literal(value, datatype=XSD.dateTime)
+
+    def _cast_num(self, value):
+        formats = [
+            "%Y-%m-%d %H:%M:%S",
+            "%Y-%m-%d %H:%M",
+            "%Y-%m-%d %H",
+            "%Y-%m-%d",
+            "%Y-%m",
+            "%Y",
+        ]
+        date = None
+        for fmt in formats:
```

Comment:
One nitpick: this is gonna get run millions upon millions of times. I'm wondering it it would be worthwhile putting the formats into a class-level dict that's indexed by string length, i.e.: `{4: '%Y'}`. That way, rather than trying each format, the correct one could be inferred in O(1) and only one call to `strptime` is required.